actionpack/lib/action_dispatch/routing/inspector.rb

```diff
@@ -223,9 +223,16 @@ def no_routes(routes)
         private
 
           def draw_expanded_section(routes)
+            require "io/console/size"
+            console_width = IO.console_size.second
+
             routes.map.each_with_index do |r, i|
+              section_name = "--[ Route #{i + 1} ]"
+              line_width = console_width - section_name.size
+              line_width = 0 if line_width.negative?
```

Comment:
This seems like it's working at a separate level of abstraction. How about extracting it to a `route_header` method: `route_header(index: i + 1)`?

```ruby
def route_header(index:)
  @console_width ||= begin
    require "io/console/size"
    IO.console_size.second
  end

  header_prefix     = "--[ Route #{index} ]"
  dash_remainder = [ console_width - header_prefix.size, 0 ].max
  
  "#{header_prefix}#{"-" * dash_remainder}"
end
```

Reply:
Looks like we can omit ivar.
```diff
diff --git a/actionpack/lib/action_dispatch/routing/inspector.rb b/actionpack/lib/action_dispatch/routing/inspector.rb
index 19dfb44..40fd007 100644
--- a/actionpack/lib/action_dispatch/routing/inspector.rb
+++ b/actionpack/lib/action_dispatch/routing/inspector.rb
@@ -102,6 +102,7 @@ def filter_routes(filter)
         end

         def collect_routes(routes)
+          Array.new(1000,
           routes.collect do |route|
             RouteWrapper.new(route)
           end.reject(&:internal?).collect do |route|
@@ -112,6 +113,7 @@ def collect_routes(routes)
               path: route.path,
               reqs: route.reqs }
           end
+          ).flatten
         end

         def collect_engine_routes(route)
```
```bash
blog$ time rails routes --expanded
(snip)
--[ Route 13000 ]-----------------------------------------------------------------------------------------------------------
Prefix            | rails_direct_uploads
Verb              | POST
URI               | /rails/active_storage/direct_uploads(.:format)
Controller#Action | active_storage/direct_uploads#create

real    0m1.483s
user    0m1.037s
sys     0m0.337s
```

<hr/>

```diff
diff --git a/actionpack/lib/action_dispatch/routing/inspector.rb b/actionpack/lib/action_dispatch/routing/inspector.rb
index 19dfb44..4536a52 100644
--- a/actionpack/lib/action_dispatch/routing/inspector.rb
+++ b/actionpack/lib/action_dispatch/routing/inspector.rb
@@ -102,6 +102,7 @@ def filter_routes(filter)
         end

         def collect_routes(routes)
+          Array.new(1000,
           routes.collect do |route|
             RouteWrapper.new(route)
           end.reject(&:internal?).collect do |route|
@@ -112,6 +113,7 @@ def collect_routes(routes)
               path: route.path,
               reqs: route.reqs }
           end
+          ).flatten
         end

         def collect_engine_routes(route)
@@ -189,6 +191,7 @@ def widths(routes)
       class Expanded
         def initialize
           @buffer = []
+          @console_width = IO.console_size.second
         end

         def result
@@ -236,9 +239,8 @@ def draw_expanded_section(routes)
           end

           def route_header(index:)
-            console_width = IO.console_size.second
             header_prefix = "--[ Route #{index} ]"
-            dash_remainder = [console_width - header_prefix.size, 0].max
+            dash_remainder = [@console_width - header_prefix.size, 0].max

             "#{header_prefix}#{'-' * dash_remainder}"
           end
```

```bash
blog$ time rails routes --expanded
(snip)
--[ Route 13000 ]-----------------------------------------------------------------------------------------------------------
Prefix            | rails_direct_uploads
Verb              | POST
URI               | /rails/active_storage/direct_uploads(.:format)
Controller#Action | active_storage/direct_uploads#create

real    0m1.485s
user    0m1.107s
sys     0m0.269s

```